src/index.tsx

```diff
@@ -2322,19 +2322,58 @@ export class App extends React.Component<any, AppState> {
 const rootElement = document.getElementById("root");
 
 class TopErrorBoundary extends React.Component {
-  state = { hasError: false, stack: "", localStorage: "" };
+  state: {
+    unresolvedError: Error | null;
+    hasError: boolean;
+    stack: string;
+    localStorage: string;
+  } = {
+    unresolvedError: null,
+    hasError: false,
+    stack: "",
+    localStorage: "",
+  };
 
-  static getDerivedStateFromError(error: any) {
-    console.error(error);
+  static getDerivedStateFromError(error: Error) {
+    const _localStorage: any = {};
+    for (const [key, value] of Object.entries({ ...localStorage })) {
+      try {
+        _localStorage[key] = JSON.parse(value);
+      } catch (err) {
+        _localStorage[key] = value;
+      }
+    }
     return {
       hasError: true,
-      localStorage: JSON.stringify({ ...localStorage }),
-      stack: error.stack,
+      unresolvedError: error,
+      localStorage: JSON.stringify(_localStorage),
     };
   }
 
+  async componentDidUpdate() {
+    if (this.state.unresolvedError !== null) {
+      const StackTrace = await import("stacktrace-js");
```

Comment:
Should this be in the `try` block?

Reply:
Dunno, should we expect our network to fail?

Reply:
Would have to put it inside a loop in the next commit. It probably won't have perf cost though, will check.

Reply:
Seems fine.